src/pip/_internal/req/req_install.py

```diff
@@ -658,13 +661,55 @@ def run_egg_info(self):
             metadata_name = canonicalize_name(self.metadata["Name"])
             if canonicalize_name(self.req.name) != metadata_name:
                 logger.warning(
-                    'Running setup.py (path:%s) egg_info for package %s '
+                    'Generating metadata for package %s '
                     'produced metadata for project name %s. Fix your '
                     '#egg=%s fragments.',
-                    self.setup_py, self.name, metadata_name, self.name
+                    self.name, metadata_name, self.name
                 )
                 self.req = Requirement(metadata_name)
 
+    def prepare_pep517_metadata(self):
+        assert self.pep517_backend is not None
+
+        metadata_dir = os.path.join(
+            self.setup_py_dir,
+            'pip-wheel-metadata'
```

Comment:
Having played around a bit, perhaps we should use a different name here: pip-wheel-metadata stands out... Perhaps a `.` at the start and perhaps even a sub-folder `.pip-metadata/wheel` would be a good idea?

Idk. It's a little subjective but I expect this to be something that shows up in development constantly so we should try to optimize the ergonomics here.

Reply:
Perhaps `build/pip-wheel-metadata`?